Sources/CartonHelpers/DiagnosticsParser.swift

```diff
@@ -0,0 +1,207 @@
+// Copyright 2020 Carton contributors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import Foundation
+import TSCBasic
+import Splash
+
+fileprivate extension StringProtocol {
+  func matches(regex: NSRegularExpression) -> String.SubSequence? {
+    let str = String(self)
+    let range = NSRange(location: 0, length: utf16.count)
+    guard let match = regex.firstMatch(in: str, options: [], range: range),
+          let matchRange = Range(match.range, in: String(self)) else {
+      return nil
+    }
+    return str[matchRange.upperBound..<str.endIndex]
+  }
+}
+
+fileprivate extension String.StringInterpolation {
+  mutating func appendInterpolation<T>(_ value: T, color: String...) {
+    appendInterpolation("\(color.map { "\u{001B}\($0)" }.joined())\(value)\u{001B}[0m")
+  }
+}
+
+fileprivate extension TokenType {
+  var color: String {
+    switch self {
+    case .keyword: return "[35;1m"
+    case .comment: return "[90m"
+    case .call, .dotAccess, .property, .type: return "[94m"
+    case .number, .preprocessing: return "[33m"
+    case .string: return "[91;1m"
+    default: return "[0m"
+    }
+  }
+}
+
+fileprivate struct TerminalOutputFormat: OutputFormat {
+  func makeBuilder() -> TerminalOutputBuilder {
+    .init()
+  }
+  
+  struct TerminalOutputBuilder: OutputBuilder {
+    var output: String = ""
+    
+    mutating func addToken(_ token: String, ofType type: TokenType) {
+      output.append("\(token, color: type.color)")
+    }
+    
```

Comment:
<!--
  1 failure:  Lines should not ...
  0 warning: 
  
  
  DangerID: danger-id-Danger;
  File: Sources/CartonHelpers/DiagnosticsParser.swift;
  Line: 61;
-->
- :no_entry_sign: Lines should not have trailing whitespace. (`trailing_whitespace`)